core/src/main/java/com/linecorp/armeria/internal/common/DefaultRequestTarget.java

```diff
@@ -443,6 +451,7 @@ private static RequestTarget slowForServer(String reqTarget, boolean allowSemico
                                         -1,
                                         matrixVariablesRemovedPath,
                                         encodedPath,
+                                        reqTarget,
```

Comment:
my read of the code is that the reqTarget here is the original one in Http2RequestDecoder but it actually may be modified at Http1RequestDecoder.